frontend/src/components/Map/Map.js

```diff
@@ -57,15 +59,14 @@ function Map({
   const [mapLoaded, setMapLoaded] = useState(false)
   const [metricConfig, setMetricConfig] = useState(null)
   const [colorScheme, setColorScheme] = useState([])
-  const [, setLayerIds] = useState([])
-  const [, setSourceIds] = useState([])
   const [popupData, setPopupData] = useState(null)
+  const sourceIds = useRef([])
 
   const domain = useMemo(() => metricConfig?.layers?.[0]?.metric?.domain || [], [metricConfig?.layers])
 
   const initPopup = useCallback(
     ({ foreignKey, metricKey, titleKey, layerId, sourceLayer }) => {
-      const sourceName = `source_${metricConfig?.id || ''}`
+      const sourceName = `source_${metricConfig?.geometry?.promoteId || ''}`
```

Comment:
We might end up with two sources in the future with the same `promoteId` but a different `url`. Maybe make this a hash of `url` and `promoteId`? Or something else that varies with both of those variables.

Same goes for other lines which generate the source name in this file.

Reply:
@justinanderson yes, I was wondering between url and promoteId but thought both will be unique. Will change to url hash